spec/controllers/rooms_controller_spec.rb

```diff
@@ -180,6 +180,22 @@ def random_valid_room_params
       expect(response).to redirect_to(r)
     end
 
+    it "should respond with JSON object of the room_settings" do
+      @request.session[:user_id] = @owner.id
+      name = Faker::Games::Pokemon.name
+
+      room_params = { name: name, "mute_on_join": "1",
+        "require_moderator_approval": "1", "anyone_can_start": "1", "all_join_moderator": "1" }
+
+      json_room_settings = "{\"muteOnStart\":true,\"requireModeratorApproval\":true," \
+        "\"anyoneCanStart\":true,\"joinModerator\":true}"
+
+      post :create, params: { room: room_params }
```

Comment:
You should try avoiding make 2 HTTP requests in a single test case. If you want to create a room, you can use the room factory (in `factories.rb`). Once you create the room, you can set the room_settings on that object